tesseract_task_composer/core/include/tesseract_task_composer/core/nodes/sync_task.h

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file sync_task.h
+ *
+ * @author Levi Armstrong
+ * @date August 13, 2023
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2023, Plectix Robotics
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_TASK_COMPOSER_SYNC_TASK_H
+#define TESSERACT_TASK_COMPOSER_SYNC_TASK_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <boost/serialization/access.hpp>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_task_composer/core/task_composer_task.h>
+
+namespace tesseract_planning
+{
+class TaskComposerPluginFactory;
+
+/**
+ * @brief This is used as a syncronization point in your task graph
+ * This task has no inputs or output and is not conditional
+ */
+class SyncTask : public TaskComposerTask
```

Comment:
What is the point of this task if it doesn't actually do anything and doesn't take inputs, set outputs?

Reply:
It is a synchronization point.

Reply:
How exactly is it used? To make sure parallel tasks start at the same time or something? I don't fully understand why that is needed and I don't see it being used anywhere.

Reply:
It would replace the StartTask as a general synchronization task.

Reply:
How does it synchronize? I don't see anything in the class itself that makes sure all of its input tasks are complete before setting success. Is that something that taskflow is actually doing behind the scenes?

Reply:
It is something that the executor will inherently do based on the graph structure but it needs a actual task so you can connect all parallel tasks to so when it gets to this task the executor will not proceed until all preceding tasks are finished.

Reply:
Makes sense; we should add something like that comment to the doc string of the class. It wasn't totally clear to me why this should work. It's also a little unclear that not accepting input/output data does not mean it cannot accept input/output connections to other tasks

Reply:
I updated the documentation of the class.